core/src/main.cpp

```diff
@@ -651,21 +676,38 @@ auto set_log_level(const json& _config) -> void
 
 auto init_tls(const json& _config) -> void
 {
-	const auto set_env_var = [&](const auto& _json_ptr_path, const char* _env_var, const char* _default_value = "") {
-		using json_ptr = json::json_pointer;
+	const auto set_env_var_string =
+		[&](const auto& _json_ptr_path, const char* _env_var, const char* _default_value = "") {
+			using json_ptr = json::json_pointer;
+
+			if (const auto v = _config.value(json_ptr{_json_ptr_path}, _default_value); !v.empty()) {
+				const auto env_var_upper = boost::to_upper_copy<std::string>(_env_var);
+				log::trace("Setting environment variable [{}] to [{}].", env_var_upper, v);
+				setenv(env_var_upper.c_str(), v.c_str(), 1); // NOLINT(concurrency-mt-unsafe)
+			}
+		};
 
-		if (const auto v = _config.value(json_ptr{_json_ptr_path}, _default_value); !v.empty()) {
-			const auto env_var_upper = boost::to_upper_copy<std::string>(_env_var);
-			log::trace("Setting environment variable [{}] to [{}].", env_var_upper, v);
-			setenv(env_var_upper.c_str(), v.c_str(), 1); // NOLINT(concurrency-mt-unsafe)
-		}
+	const auto set_env_var_int = [&](const auto& _json_ptr_path, const char* _env_var, int _default_value) {
```

Comment:
Same as other capture comment:

Just to be sure, the capture only takes `_config` right?

Reply:
Good catch.

Will lock that down.

Reply:
Done.